crates/bevy_reflect/src/array.rs

```diff
@@ -8,16 +8,40 @@ use std::{
     hash::{Hash, Hasher},
 };
 
-/// A static-sized array of [`Reflect`] items.
+/// A trait used to power [array-like] operations via [reflection].
 ///
-/// This corresponds to types like `[T; N]` (arrays).
+/// This corresponds to true Rust arrays like `[T; N]`,
+/// but also to any fixed-size linear sequence types.
+/// It also serves as a supertrait for the [`List`] trait.
```

Comment:
Yeah it might be worthwhile to indicate that this behavior is not strictly enforced and that implementors are strongly advised to adhere to this contract.

However, we might not need to mention `List` if #7467 gets merged.